### PR TITLE
LPS-162139 Adapt the CustomView Filter Creation to support relationship fields

### PR DESCRIPTION
**Pull request by p-albuquerque**
Related issue: https://issues.liferay.com/browse/LPS-162139

### CODE REVIEW

**Comment by p-albuquerque** [comment]
ci:test:relevant

File: modules/apps/object/object-service/src/main/java/com/liferay/object/internal/field/filter/parser/ListObjectFieldFilterParser.java [hunk 1]
```diff
@@ -139,7 +153,56 @@ private List<Integer> _toIntegerList(JSONArray jsonArray) {
 		return statuses;
 	}
 
+	private void _validate(
+			JSONArray jsonArray, long objectDefinitionId,
+			String objectFieldName)
+		throws PortalException {
+
+		ObjectEntry objectEntry;
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/internal/field/filter/parser/ListObjectFieldFilterParser.java:173]: `ObjectEntry objectEntry = null;`

File: modules/apps/object/object-service/src/main/java/com/liferay/object/internal/field/filter/parser/ListObjectFieldFilterParser.java [hunk 2]
```diff
@@ -139,7 +153,56 @@ private List<Integer> _toIntegerList(JSONArray jsonArray) {
...
+
+		for (int i = 0; i < jsonArray.length(); i++) {
+			try {
+				objectEntry = _objectEntryLocalService.getObjectEntry(
+					Long.parseLong((String)jsonArray.get(i)));
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/internal/field/filter/parser/ListObjectFieldFilterParser.java:178]: just cast to Long

File: modules/apps/object/object-service/src/main/java/com/liferay/object/internal/field/filter/parser/ListObjectFieldFilterParser.java [hunk 3]
```diff
@@ -139,7 +153,56 @@ private List<Integer> _toIntegerList(JSONArray jsonArray) {
...
+			}
+			catch (Exception exception) {
+				throw new ObjectViewFilterColumnException(
+					StringBundler.concat(
+						"Value: ", jsonArray.get(i), "is not ObjectEntryId"),
+					exception);
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/internal/field/filter/parser/ListObjectFieldFilterParser.java:184]: This ID does not belong to an ObjectEntry.

**Review by MarinhoFeliphe — CHANGES_REQUESTED** [review]

**Review by MarinhoFeliphe — CHANGES_REQUESTED** [review]

**Comment by p-albuquerque** [comment]
Working on additional stuffs that will be increased on it

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Comment by p-albuquerque** [comment]
ci:test:sf

**Comment by p-albuquerque** [comment]
ci:test:sf

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Comment by p-albuquerque** [comment]
ci:test:sf

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Comment by naraandrade** [comment]
@p-albuquerque Could you please check if the failures are related? Thanks :)

**Comment by p-albuquerque** [comment]
ci:test:relevant

**Comment by MarinhoFeliphe** [comment]
Resent here: https://github.com/liferay-objects/liferay-portal/pull/1112

**Pull request closed without merging**